tests/test_other.py

```diff
@@ -7760,18 +7760,25 @@ def build():
     self.assertIdentical(normal, tiny)
     self.assertIdentical(normal, huge)
 
-  def test_EM_ASM_ES6(self):
+  @parameterized({
+    '': ([],), # noqa
+    'O3': (['-O3'],), # noqa
+    'closure': (['--closure', '1'],), # noqa
+    'closure_O3': (['--closure', '1', '-O3'],), # noqa
+  })
+  def test_EM_ASM_ES6(self, args):
     create_test_file('src.cpp', r'''
 #include <emscripten.h>
 int main() {
   EM_ASM({
-    var x = (a, b) => 5; // valid ES6
+    let x = (a, b) => 5; // valid ES6
     async function y() {} // valid ES2017
```

Comment:
may also want to ensure y isn't DCE'd away. Can do `Module.save_y = y;` for example.